src/libs/OptionsListUtils.ts

```diff
@@ -1184,6 +1222,12 @@ function getValidOptions(
         return true;
     });
 
+    let workspaceChats: ReportUtils.OptionData[] = [];
+
+    if (shouldSeparateWorkspaceChat) {
+        workspaceChats = allReportOptions.filter((option) => option.isPolicyExpenseChat);
```

Comment:
I think we should build new workspaces OptionData from [allPolicies](https://github.com/callstack-internal/Expensify-App/blob/main/src/libs/OptionsListUtils.ts#L214) using the same [filter conditions](https://github.com/callstack-internal/Expensify-App/blob/main/src/pages/workspace/WorkspacesListPage.tsx#L343) as those applied on the workspace list page. <!-- This approach ensures consistency and accurate data representation across both views. -->

https://github.com/user-attachments/assets/1758f438-fc60-481b-8f7a-80e95d0941e5

Reply:
Can you specify what exactly you mean by that? And where exactly would you like this to be implemented?

Reply:
> Can you specify what exactly you mean by that? And where exactly would you like this to be implemented?

Based on the current implementation, the workspace list does not include all workspaces, and it still cannot retrieve my primary workspace. Therefore, we need workspace options that are the same as the `WorkspacesListPage`. You can see the difference between ws options with the workspace list in this video.

https://github.com/user-attachments/assets/1758f438-fc60-481b-8f7a-80e95d0941e5

> Do you think we should return another key in createOptionList()? (e.g. workspaces with all workspace chats)

I think we can create new function `type TCreateWorkspaceOptions = () => ReportUtils.OptionData[]` to build all ws options.

Reply:
@trjExpensify Could you please help me confirm if the workspace options in this issue will be the same as those in the `WorkspacesListPage`?

Reply:
Hm, not necessarily. We only show workspaces with `isPolicyExpenseChatEnabled` in the expense flow. The workspace must have a workspace chat to be able to submit the expense to. 

I do know that we considered updating the Workspaces page list to hide any that don't have `isPolicyExpenseChatEnabled` as well, but I'm not sure if that materialised. CC: @puneetlath

Reply:
I can't remember if we did either unfortunately.

Reply:
No, doesn't look like we did that yet. 

<img width="1511" alt="image" src="https://github.com/user-attachments/assets/fb1a4242-0cd2-4a81-ad09-325cadf99f06" />
<img width="1511" alt="image" src="https://github.com/user-attachments/assets/f23d48f0-4254-4727-a252-3a31ceadc4f2" />

Reply:
So, should we stick with the current solution (show the ones with `isPolicyExpenseChatEnabled`) or should I rewrite it to show all workspaces? 😄

Reply:
Only show the ones with `isPolicyExpenseChatEnabled`.

Reply:
Is the current implementation ok? Or is there something that you'd like me to change about it?